### PR TITLE
feat(FEV-1164): Do request + ready promise

### PR DESCRIPTION
**Pull request by semarche-kaltura**
1) fetch userEntry data
2) fetch quiz data

### CODE REVIEW

File: src/ivq.tsx [hunk 1]
```diff
@@ -12,19 +14,53 @@ export class Ivq extends KalturaPlayer.core.BasePlugin {
...
+
+  private _getQuiz() {
+    this._player.provider
+    .doRequest([{loader: QuizLoader, params: {entryId: this._player.sources.id}}])
+    .then((data: Map<string, any>) => {
+      if (data && data.has(QuizLoader.id)) {
+        const quizLoader = data.get(QuizLoader.id);
+        const userEntry = quizLoader?.response?.userEntries[0];
+        if (userEntry) {
+          console.log("> userEntry", userEntry) // TODO: use userEntry data
```
Comment (RoyBregman) [inline src/ivq.tsx:52]: use a logger

File: src/ivq.tsx [hunk 2]
```diff
@@ -12,19 +14,53 @@ export class Ivq extends KalturaPlayer.core.BasePlugin {
...
+        }
+        const quizData = quizLoader?.response?.quiz;
+        if (quizData) {
+          console.log("> quizData", quizData)  // TODO: use quiz data
+        }
```
Comment (RoyBregman) [inline src/ivq.tsx:57]: Maybe add some warn log if no response

**Review by RoyBregman — COMMENTED** [review]
Besides my cosmetic comments LGTM

File: src/ivq.tsx (hunk 1 above)
Comment (semarche-kaltura) [reply src/ivq.tsx:52]: it's temporary code. I don't think we should add logger here and then remove it in next PR.

File: src/ivq.tsx [hunk 3]
```diff
@@ -1,30 +1,69 @@
...
 
   get ready() {
-    return Promise.resolve();
+    return this._readyPromise;
   }
 
-  loadMedia(): void {}
+  loadMedia(): void {
+    const kalturaCuePointService: any = this._player.getService('kalturaCuepoints');
+    this._getQuestionsAndAnswers(kalturaCuePointService);
```
Comment (RoyBregman) [inline src/ivq.tsx:24]: Please add some check the kalturaCuePointService exists and if not return before _getQuestionsAndAnswers an warn to console

File: src/ivq.tsx [hunk 4]
```diff
@@ -1,30 +1,69 @@
...
 
   destroy(): void {}
+
+  private _makeReadyPromise = () => {
+    return new Promise<void>(res => {
+      this._resolveReadyPromise = res;
+    });
+  };
+
+  private _getQuestionsAndAnswers(kalturaCuePointService: any) {
```
Comment (RoyBregman) [inline src/ivq.tsx:44]: I think we need to make some change in name convention so it is more clear - maybe call the meta data answers optionalAnswers to separate them from the user answered ones.

File: src/ivq.tsx [hunk 5]
```diff
@@ -1,30 +1,69 @@
...
+          if (!quizData) {
+            this.logger.warn('quiz data absent');
+          } else {
+            // TODO: use quiz data
+          }
+          this._resolveReadyPromise();
+        }
+      })
+      .catch((e: any) => {
+        this.logger.error(e);
```
Comment (RoyBregman) [inline src/ivq.tsx:65]: Please check in other repos but I think error is saved for playback severe errors.
Maybe warn is enough

File: src/providers/response-types/kaltura-quiz-answer-response.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import {KalturaQuizAnswer} from './kaltura-quiz-answer';
+const {ServiceResult} = KalturaPlayer.providers;
+
+export class KalturaQuizAnswerResponse extends ServiceResult {
```
Comment (RoyBregman) [inline src/providers/response-types/kaltura-quiz-answer-response.ts:4]: Maybe
```suggestion
export class KalturaQuizAnswerListResponse extends ServiceResult {
```

File: src/providers/quiz-loader.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    const userEntryListResponse = new KalturaUserEntryListResponse(response[0]?.data);
+    if (userEntryListResponse.totalCount) {
+      this._response.userEntries = userEntryListResponse?.data;
+    }
+    const quizResponse = new KalturaQuizResponse(response[1]?.data);
+    if (quizResponse) {
+      this._response.quiz = quizResponse?.data;
+    }
+    const quizAnswersResponse = new KalturaQuizAnswerResponse(response[2]?.data);
+    if (quizAnswersResponse) {
```
Comment (RoyBregman) [inline src/providers/quiz-loader.ts:83]: Don't we need to check the totalCount instead? Like we do here:
https://github.com/kaltura/playkit-js-ivq/pull/3/files#diff-0c80c2011b14e9f2d68e4fe38d107b2a1cb01babaa484c5650c7ef642c222befR75

**Comment by RoyBregman** [comment]
Done reviewing, small comments.
I think change ivq.tsx to ts as it is not really a React component. I think it is confusing and relevant

**Comment by semarche-kaltura** [comment]
> Done reviewing, small comments. I think change ivq.tsx to ts as it is not really a React component. I think it is confusing and relevant

done

**Review by RoyBregman — APPROVED** [review]

**Pull request merged**